README.md

```diff
@@ -3,3 +3,89 @@
 This package binds to Cronet's [native API](https://chromium.googlesource.com/chromium/src/+/master/components/cronet/native/test_instructions.md) to expose them in Dart.
 
 This is a [GSoC 2021 project](https://summerofcode.withgoogle.com/projects/#4757095741652992).
+
+## Usage
+
+1. Add package as a dependency in your `pubspec.yaml`.
+
+2. Run this from the `root` of your project.
+
+   **Dart CLI**
+
+   ```bash
+   dart pub get
+   dart run cronet <platform> # Downloads the cronet binaries.
```

Comment:
This doesn't work for me.

```
dacoharkes@dacoharkes-l:~/cronet.dart$ dart run cronet linux64
Building Wrapper...
Unhandled exception:
ProcessException: Permission denied
  Command: /usr/local/google/home/dacoharkes/cronet.dart/lib/src/native/wrapper/build.sh /usr/local/google/home/dacoharkes/cronet.dart/lib/src/native/wrapper "86.0.4240.198"
#0      _ProcessImpl._runAndWait (dart:io-patch/process_patch.dart:476:7)
#1      _runNonInteractiveProcessSync (dart:io-patch/process_patch.dart:619:18)
#2      Process.runSync (dart:io-patch/process_patch.dart:66:12)
#3      buildWrapper (package:cronet/src/prepare_cronet.dart:33:15)
#4      main (file:///usr/local/google/home/dacoharkes/cronet.dart/bin/cronet.dart:10:7)
#5      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:281:32)
#6      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)

$ /usr/local/google/home/dacoharkes/cronet.dart/lib/src/native/wrapper/build.sh /usr/local/google/home/dacoharkes/cronet.dart/lib/src/native/wrapper "86.0.4240.198"
-bash: /usr/local/google/home/dacoharkes/cronet.dart/lib/src/native/wrapper/build.sh: Permission denied
```

The file is not marked as executable.

`-rw-r----- 1 dacoharkes primarygroup  459 Jun 16 10:13 build.sh`

```
$ bash /usr/local/google/home/dacoharkes/cronet.dart/lib/src/native/wrapper/build.sh /usr/local/google/home/dacoharkes/cronet.dart/lib/src/native/wrapper "86.0.4240.198"
In file included from wrapper.cc:9:
<command-line>: error: too many decimal points in number
wrapper.h:19:36: note: in definition of macro ‘OPENLIB’
   19 |   #define OPENLIB(libname) dlopen((libname), RTLD_NOW)
      |                                    ^~~~~~~
wrapper.cc:18:49: note: in expansion of macro ‘CRONET_VERSION’
   18 |   #define CRONET_LIB_NAME CRONET_LIB_PREFIX "." CRONET_VERSION CRONET_LIB_EXTENSION
      |                                                 ^~~~~~~~~~~~~~
wrapper.cc:40:26: note: in expansion of macro ‘CRONET_LIB_NAME’
   40 | LIBTYPE handle = OPENLIB(CRONET_LIB_NAME);
      |                          ^~~~~~~~~~~~~~~
<command-line>: error: expected ‘)’ before numeric constant
wrapper.h:19:36: note: in definition of macro ‘OPENLIB’
   19 |   #define OPENLIB(libname) dlopen((libname), RTLD_NOW)
      |                                    ^~~~~~~
wrapper.cc:18:49: note: in expansion of macro ‘CRONET_VERSION’
   18 |   #define CRONET_LIB_NAME CRONET_LIB_PREFIX "." CRONET_VERSION CRONET_LIB_EXTENSION
      |                                                 ^~~~~~~~~~~~~~
wrapper.cc:40:26: note: in expansion of macro ‘CRONET_LIB_NAME’
   40 | LIBTYPE handle = OPENLIB(CRONET_LIB_NAME);
      |                          ^~~~~~~~~~~~~~~
wrapper.h:19:35: note: to match this ‘(’
   19 |   #define OPENLIB(libname) dlopen((libname), RTLD_NOW)
      |                                   ^
wrapper.cc:40:18: note: in expansion of macro ‘OPENLIB’
   40 | LIBTYPE handle = OPENLIB(CRONET_LIB_NAME);
      |                  ^~~~~~~
wrapper.cc:40:42: error: expected ‘)’ before ‘;’ token
   40 | LIBTYPE handle = OPENLIB(CRONET_LIB_NAME);
      |                                          ^
In file included from wrapper.cc:9:
wrapper.h:19:34: note: to match this ‘(’
   19 |   #define OPENLIB(libname) dlopen((libname), RTLD_NOW)
      |                                  ^
wrapper.cc:40:18: note: in expansion of macro ‘OPENLIB’
   40 | LIBTYPE handle = OPENLIB(CRONET_LIB_NAME);
      |                  ^~~~~~~
```

Reply:
Try running `./build.sh . '"86.0.4240.198"'`. This may work :sweat_smile: But, we're dartifying it anyways.

Reply:
Replaced build.sh script with dart implementation.

Reply:
You can now run the build script as `build_cronet <version>`. Version string is optional. Defaults to: `86.0.4240.198`. Activate the package, before doing `build_cronet,` using `dart pub global activate`